test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
So I ended up using `tcgetattr` to populate the `Termios`. Apparently this breaks on OS X because they don't support `tcgetattr`?